source/common/http/match_delegate/config.cc

```diff
@@ -96,9 +100,17 @@ struct DelegatingFactoryCallbacks : public Envoy::Http::FilterChainFactoryCallba
 
 void DelegatingStreamFilter::FilterMatchState::evaluateMatchTree(
     MatchDataUpdateFunc data_update_func) {
-  if (!has_match_tree_ || match_tree_evaluated_) {
+  if (match_tree_evaluated_) {
     return;
   }
+
+  // If no match tree is set, interpret as a skip.
+  if (!has_match_tree_) {
+    skip_filter_ = true;
```

Comment:
this wont invoke the on_no_match right?

Reply:
This will not invoke the on_no_match. I believe that in order for the on_no_match to be used it has to be included in a matcher. This code is triggered when there is no matcher set at all.